main.py

```diff
@@ -4,40 +4,53 @@
 import sys
 import json
 import queue
+import logging
 from threading import Thread
 
 APPLICATION_PORT = 65412
 message_queue = queue.Queue()
+logger = logging.getLogger(__name__)
+logging.basicConfig(filename=f"chat.log", level=logging.DEBUG, format="%(asctime)s - %(message)s")
 
 def ui(peer_host, peer_port, input=input, socket=os_socket):
-    while True:
-        message = input("viestisi: ")
-        with socket(AF_INET, SOCK_STREAM) as s:
-            s.connect((peer_host, peer_port))
-            s.sendall(json.dumps({"message": message}).encode())
-            data = s.recv(1024)
+    try:
+        while True:
+            message = input("viestisi: ")
+            with socket(AF_INET, SOCK_STREAM) as s:
+                s.connect((peer_host, peer_port))
+                s.sendall(json.dumps({"message": message}).encode())
+                data = s.recv(1024)
 
-            print()
-            print(f"Received {data!r}")
+                print()
+                print(f"Received {data!r}")
+                logger.debug(f"Sent by {peer_host}: {data}")
+    except Exception as exc:
+        logger.exception(exc)
+        raise exc
 
-def start_server(socket=os_socket):
-    with socket(AF_INET, SOCK_STREAM) as s:
-        s.bind(("0.0.0.0", APPLICATION_PORT))
-        s.listen()
-        while True:
-            conn, addr = s.accept()
-            with conn:
-                while True:
-                    data = conn.recv(1024)
-                    if not data:
-                        break
-                    print()
-                    print(f"Received by {data}")
-                    message_queue.put(data)
-
-                    if not message_queue.empty():
-                        message = message_queue.get()
-                        conn.sendall(message)
+def start_server(peer_host, socket=os_socket):
+    try:
+        with socket(AF_INET, SOCK_STREAM) as s:
+            s.bind(("0.0.0.0", APPLICATION_PORT))
+            s.listen()
+            while True:
+                conn, addr = s.accept()
+                with conn:
+                    while True:
+                        data = conn.recv(1024)
+                        if not data:
+                            break
+                        print()
+                        print(f"Received by {data}")
+                        logger.debug(f"Received by {peer_host}: {data}")
```

Comment:
The s.accept returns the ip address of the peer. Now the system would only support one peer connection